### ISSUE TITLE
Persistent Invalid Cloud Messaging Token Returned on iOS

### ISSUE THREAD

**Issue by timlajaunie** [TobiasBuchholz/Plugin.Firebase#324]
Hi.  We're using Plugin.Firebase.CloudMessaging version 3.0.0 in our .NET 8 MAUI project.

We noticed the cloud messaging token is always the same, invalid token on iOS, even after uninstalling/reinstalling the app.  When attempting to send notifications to the provided token, we receive an "UNREGISTERED" error code from the FCM API.

We've thoroughly read through all the setup documentation on this repo and [Adam E's GoogleApisForiOSComponents](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents) repo.

Our project follows [andyzukunft's sample](https://github.com/andyzukunft/Plugin.Firebase/tree/fcm-demo/sample/Fcm) linked in the [troubleshooting section](https://github.com/TobiasBuchholz/Plugin.Firebase/blob/development/docs/cloud_messaging.md#troubleshooting)  of the cloud messaging documentation.  That sample appears to be somewhat outdated now, and references Plugin.Firebase.CloudMessaging version 2.0.3, but there does not appear to be a more relevant, recent sample to use as a guide.  We even adapted the sample to .NET 8 and version 3.0.0 of the NuGet package, and we observed the same behavior.

To be clear, we are able to successfully publish an ad-hoc iOS build and deploy to a physical device.  The issue is simply the persistent invalid token being returned on iOS.  Additionally, published Android builds work fine on the emulator and physical devices.

Any ideas about this issue, or tips to get iOS working correctly so we can finally put this to bed?

**Comment by dartasen** [comment]
@timlajaunie I might be wrong but looking at your [Info.plist](https://github.com/andyzukunft/Plugin.Firebase/blob/fcm-demo/sample/Fcm/Platforms/iOS/Info.plist), I think you might need to include your URL scheme from your firebase apple app :

Got something working like a charm with this :
```xml
<key>CFBundleURLTypes</key>
<array>
    <dict>
	    <key>CFBundleURLName</key>
	    <string>CFBundleURLName</string>
	    <key>CFBundleTypeRole</key>
	    <string>Editor</string>
	    <key>CFBundleURLSchemes</key>
	    <array>
		    <string>com.google.fcm</string>
		    <string>app-1-[REDACTED]-ios-[REDACTED]</string>
	    </array>
    </dict>
</array>
```

**Comment by timlajaunie** [comment]
> @timlajaunie I might be wrong but looking at your [Info.plist](https://github.com/andyzukunft/Plugin.Firebase/blob/fcm-demo/sample/Fcm/Platforms/iOS/Info.plist), I think you might need to include your URL scheme from your firebase apple app :
> 
> Got something working like a charm with this :
> 
> ```
> <key>CFBundleURLTypes</key>
> <array>
>     <dict>
> 	    <key>CFBundleURLName</key>
> 	    <string>CFBundleURLName</string>
> 	    <key>CFBundleTypeRole</key>
> 	    <string>Editor</string>
> 	    <key>CFBundleURLSchemes</key>
> 	    <array>
> 		    <string>com.google.fcm</string>
> 		    <string>app-1-[REDACTED]-ios-[REDACTED]</string>
> 	    </array>
>     </dict>
> </array>
> ```

Thanks for the reply.  The Info.plist you linked is actually in the sample created by andyzukunft, which is now outdated and references a 2.X version.

Were you able to get things working with the 3.0.0 version?  I'm a little confused by your mention of "app-1-[REDACTED]-ios-[REDACTED]".  Where would I find my own URL like that to include in Info.plist?

**Comment by dartasen** [comment]
@timlajaunie Yup, got it working on both Android/iOS on .NET 8 MAUI with the latest version of the plugin. In the [README](https://github.com/TobiasBuchholz/Plugin.Firebase?tab=readme-ov-file#for-windows-users) of the plugin there's a .NET 8 Cloud Messaging/Analytics : [https://github.com/coop-tim/maui-sample](https://github.com/coop-tim/maui-sample)

>  I'm a little confused by your mention of "app-1-[REDACTED]-ios-[REDACTED]". Where would I find my own URL like that to include in Info.plist?

Under the settings page of the iOS app inside firebase [https://console.firebase.google.com/project/sample-gcm-maui/settings/general/ios:com.test.maui](https://console.firebase.google.com/project/sample-gcm-maui/settings/general/ios:com.test.maui)

Should look like this, I highlighted in red the relevant URL for the Info.Plist
![image](https://github.com/user-attachments/assets/d854af2b-e86b-4b81-9c2d-d1059be8aac9)

**Comment by timlajaunie** [comment]
We have discovered the issue.  [andyzukunft's sample](https://github.com/andyzukunft/Plugin.Firebase/tree/fcm-demo/sample/Fcm) includes disabling swizzling by inclusion of the following elements in Info.plist:

	<key>FirebaseAppDelegateProxyEnabled</key>
	<false/>

This may have worked fine when targeting version 2.X of the plugin but does not seem to work with version 3.0.0 without additional logic to manually capture the APNs token and hand it over to Firebase messaging on iOS.

It is quite confusing how the official [Plugin.Firebase.CloudMessaging docs](https://github.com/TobiasBuchholz/Plugin.Firebase/blob/development/docs/cloud_messaging.md) recommend referring to andyzukunft's sample, which is outdated and will not work out of the box with version 3.0.0.

I notice coop-tim's sample is also outdated and references version 2.X of the plugin.  However, if we had used that as a guide instead of andyzukunft's sample, we wouldn't have found ourselves in this predicament.
